app-libs/stf/src/getter.rs

```diff
@@ -293,6 +298,25 @@ impl ExecuteGetter for PublicGetter {
 		match self {
 			PublicGetter::some_value => Some(42u32.encode()),
 			PublicGetter::total_issuance => Some(Balances::total_issuance().encode()),
+			PublicGetter::undistributed_fees => {
+				let pot: AccountId = enclave_signer_account();
+				let info = System::account(&pot);
+				debug!("PublicGetter undistributed_fees");
+				debug!("AccountInfo for {} is {:?}", account_id_to_string(&pot), info);
+				let fees = info.data.free;
+				// for privacy reasons, we add some noise to the fees.
+				// This avoids leaking the exact number and cost of recent TrustedCalls
+				let noise =
+					MinimalChainSpec::one_unit(shielding_target_genesis_hash().unwrap_or_default())
+						/ STF_TX_FEE_UNIT_DIVIDER
+						* Balance::from(SgxRandomness::random_u32(0, 10));
+				let noisy_fees = fees.saturating_sub(noise);
```

Comment:
the uniform distribution noise equals tx fees for 0 to 10 cheapest TrustedCalls and is subtracted from the real value.